server/typedb.rs

```diff
@@ -164,19 +164,43 @@ impl Server {
         DiagnosticsManager::new(diagnostics, config.monitoring_port, config.is_monitoring_enabled, is_development_mode)
     }
 
-    fn create_tonic_server(encryption_config: &EncryptionConfig) -> tonic::transport::Server {
-        let mut tonic_server = tonic::transport::Server::builder();
-        if encryption_config.enabled {
-            let cert = fs::read_to_string(encryption_config.cert.clone().unwrap()).unwrap();
-            let cert_key = fs::read_to_string(encryption_config.cert_key.clone().unwrap()).unwrap();
-            let mut tls_config = ServerTlsConfig::new().identity(Identity::from_pem(cert, cert_key));
-            if encryption_config.root_ca.is_some() {
-                let root_ca = fs::read_to_string(encryption_config.root_ca.clone().unwrap()).unwrap();
-                tls_config = tls_config.client_ca_root(Certificate::from_pem(root_ca)).client_auth_optional(true)
-            }
-            tonic_server = tonic_server.tls_config(tls_config).unwrap();
+    fn create_tonic_server(encryption_config: &EncryptionConfig) -> Result<tonic::transport::Server, ServerOpenError> {
+        let mut tonic_server = Self::configure_server_encryption(tonic::transport::Server::builder(), encryption_config)?;
+        Ok(tonic_server.http2_keepalive_interval(Some(GRPC_CONNECTION_KEEPALIVE)))
+    }
+
+    fn configure_server_encryption(
+        server: tonic::transport::Server,
+        encryption_config: &EncryptionConfig,
+    ) -> Result<tonic::transport::Server, ServerOpenError> {
+        if !encryption_config.enabled {
+            return Ok(server);
+        }
+
+        let cert_path = encryption_config.cert.clone().ok_or_else(|| ServerOpenError::MissingTLSCertificate {})?;
+        let cert = fs::read_to_string(cert_path.clone()).map_err(|source| ServerOpenError::CouldNotReadTLSCertificate {
+            path: cert_path.to_string_lossy().to_string(),
+            source: Arc::new(source),
+        })?;
+        let cert_key_path =
+            encryption_config.cert_key.clone().ok_or_else(|| ServerOpenError::MissingTLSCertificateKey {})?;
+        let cert_key =
+            fs::read_to_string(cert_key_path.clone()).map_err(|source| ServerOpenError::CouldNotReadTLSCertificateKey {
+                path: cert_key_path.to_string_lossy().to_string(),
+                source: Arc::new(source),
+            })?;
```

Comment:
1. `clone()`s are unnecessary here,
2. `display()` should be used instead oif `to_string_lossy()` (an argument could be made for `{:?}` debug prints as that's meant to be fully lossless)
```suggestion
        let cert_path = encryption_config.cert.as_ref().ok_or_else(|| ServerOpenError::MissingTLSCertificate {})?;
        let cert = fs::read_to_string(cert_path).map_err(|source| ServerOpenError::CouldNotReadTLSCertificate {
            path: cert_path.display().to_string(),
            source: Arc::new(source),
        })?;
        let cert_key_path =
            encryption_config.cert_key.as_ref().ok_or_else(|| ServerOpenError::MissingTLSCertificateKey {})?;
        let cert_key =
            fs::read_to_string(cert_key_path).map_err(|source| ServerOpenError::CouldNotReadTLSCertificateKey {
                path: cert_key_path.display().to_string(),
                source: Arc::new(source),
            })?;
```